feat/leaderboard_v2 setup

BlackStarkGoku opened the pull request:
Feature: Add leaderboard_v2 except top50 users view

resolves #416

BlackStarkGoku committed:
feat/leaderboard_v2 setup

ayushtom commented on types/frontTypes.d.ts:
```diff
@@ -29,7 +29,6 @@ type Task = {
   hasRootDomain: boolean;
   customError: string;
   checkUserRewards: () => void;
-  expired: boolean;
```
i think this is the issue

BlackStarkGoku committed:
feat/ Add top 50 players view

-Fix last build with fix typescript
-Display top 50 players view when not logged or in top50 tab

ayushtom commented:
Initial feedback

I think you make another api call with a increased page size when a user clicks on view more. I think a better implementation would be to -
-  Make the initial call with params -{ addr: user_address, page_size:10, shift:0, duration :'all'}
- second call when view more is clicked with params - { addr: user_address, page_size:10, shift:1, duration :'all'}

And append the response to the previous call if the view more button is clicked


Hover interaction on top leaderboard cards is missing

BlackStarkGoku committed:
update/ Change View more button logic on leaderboard

-Add new logic to fetch leaderboard data
- Update Leaderboard Cards UI

BlackStarkGoku committed:
update/ Fix bug svg

Update stopColor property of svgs

ayushtom commented:
<img width="87" alt="Screenshot 2024-04-27 at 1 40 53 PM" src="https://github.com/starknet-id/starknet.quest/assets/41674634/a9a7fbb2-dc7b-448c-9027-a71b2f9b8976">

- This logo needs to be vertically aligned along the edge. Currently it seems to be more downward than the line.

- The cursor needs to pointer but with the image we provide in the code base. You can use the code below to implement it
```
className="modified-cursor-pointer"
```

BlackStarkGoku committed:
update/ fix leaderboard cards ui

BlackStarkGoku committed:
Merge branch 'testnet' into feat/leaderboard_v2

BlackStarkGoku committed:
update/fix build

- Fix typescript errors

ayushtom commented:
https://github.com/starknet-id/starknet.quest/assets/41674634/3276ba4b-72e2-4b7b-8345-156a2fc34fba

On first load, the data for last 7 days doesn't make much sense. Later when i go through all tabs then everything is fetched correctly but on first load i can't see my profile, just the top 50. Ideally i should see my profile on this page

BlackStarkGoku committed:
update/ Fix address on initial leaderboard load

BlackStarkGoku committed:
feature/Add leaderboard link navbar dropdown

- Fix displayed leaderboard view when coming from link
- Add link button in navbar

BlackStarkGoku committed:
Merge branch 'testnet' into feat/leaderboard_v2

BlackStarkGoku committed:
update/fix ts-error

ayushtom commented:
- the search bar doesn't render the user's position in the list, only updates the percentile when my wallet is not connected
-
<img width="1033" alt="Screenshot 2024-04-30 at 1 58 28 AM" src="https://github.com/starknet-id/starknet.quest/assets/41674634/b6d5e1be-592d-46ab-a6fe-0528f707ea62">
I see a view more button when no results are found

BlackStarkGoku committed:
update/fix search leaderboard

BlackStarkGoku committed:
update/ fix fetch initial leaderboard

BlackStarkGoku committed:
update/ fix initial leaderboard results

ayushtom commented on app/leaderboard/page.tsx:
```diff
@@ -68,7 +68,13 @@ export default function Page() {
   });
+  const [inititalFetchTop50, setInititalFetchTop50] = useState(false);
 
+  const isTop50RankedView =
```
can use `useMemo` here to avoid recomputing this variable

ayushtom commented on app/leaderboard/page.tsx:
```diff
@@ -68,7 +68,13 @@ export default function Page() {
+  const isTop50RankedView =
+    !currentSearchedAddress &&
+    (duration === TOP_50_TAB_STRING || (!isConnecting && !address));
+  const isNoSearchResults =
```
same here
can be used in useMemo

ayushtom commented:
<img width="504" alt="Screenshot 2024-05-01 at 2 22 30 AM" src="https://github.com/starknet-id/starknet.quest/assets/41674634/37108355-f4a4-4cb1-808c-3b55043879f9">

- make the search bar take full width
- remove the profile icon in mobile view . I mean the "you are better" section but without the pfp if in mobile view

BlackStarkGoku committed:
update/ improve last details

BlackStarkGoku committed:
Merge branch 'testnet' into feat/leaderboard_v2

ayushtom reviewed (approved):
lgtm

ayushtom commented on app/leaderboard/page.tsx:
```diff
@@ -471,50 +543,44 @@ export default function Page() {
+              !isNoSearchResults && (
+                <Button
+                  onClick={() => {
+                    if (checkIfLastPage && !viewMore) {
```
can you put this in a onClick handler function? we avoid long inline functions in the codebase

ayushtom commented on components/leaderboard/RankingsTable.tsx:
```diff
@@ -11,21 +11,35 @@ import { useMediaQuery } from "@mui/material";
+  searchedAddress,
 }) => {
+  const { address: connectedUserAddress, isConnecting } = useAccount();
+  const isTop50RankedView =
```
useMemo here too

ayushtom commented on components/leaderboard/RankingsTable.tsx:
```diff
@@ -11,21 +11,35 @@ import { useMediaQuery } from "@mui/material";
   // used to format the data to be displayed
   const [displayData, setDisplayData] = useState<FormattedRankingProps>([]);
-
+  const slicedDisplayedData = isTop50RankedView
```
same here

ayushtom commented on components/leaderboard/RankingsTable.tsx:
```diff
@@ -66,63 +80,78 @@ const RankingsTable: FunctionComponent<RankingProps> = ({
+                  <div className={styles.ranking_table_row_name_rank}>
+                    <div className={styles.ranking_position_layout}>
+                      <p className="text-white text-center">
+                        {itemRank > 10000
```
add this in a function too called `rankFormatter`

ayushtom commented on components/leaderboard/RankingsTable.tsx:
```diff
@@ -66,63 +80,78 @@ const RankingsTable: FunctionComponent<RankingProps> = ({
+                              : "#ffffff",
+                        }}
+                      >
+                        {isMobile &&
```
useMemo this value and avoid inline longer conditions which have state variables involved. React will recompute them even if they haven't changed

ayushtom reviewed (changes requested):
final comments added

BlackStarkGoku committed:
update/ change last comments

ayushtom reviewed (approved):
lgtm
